internal/pkg/agent/application/upgrade/artifact/config.go

```diff
@@ -54,6 +54,9 @@ type Config struct {
 	// If not provided FileSystem Downloader will fallback to /beats subfolder of elastic-agent directory.
 	DropPath string `yaml:"dropPath" config:"drop_path"`
 
+	// RetryMaxCount: the maximum number of attempts to make at downloading packages.
+	RetryMaxCount int `config:"retry_max_count"`
```

Comment:
I deliberately chose to use this name for the configuration setting since we use the same name for another setting that has the same meaning:  https://github.com/elastic/elastic-agent/blob/8b8b3209ed02d2832c809e5cdd645506dacdc866/internal/pkg/agent/cmd/setup_config.go#L55